src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -166,6 +171,18 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
 
     // Continue with next recording id
     DEBUG_ASSERT(!m_queuedRecordingIds.isEmpty());
+    emit currentRecordingFetched();
+    // Some of the tracks has loads of RecordingIds.
+    // This can cause hitting the rate limits of MusicBrainz.
+    // According to the MusicBrainz API Doc: https://musicbrainz.org/doc/MusicBrainz_API/Rate_Limiting
+    // The rate limit should be one query in a second.
+    // Related Bug: https://bugs.launchpad.net/mixxx/+bug/1983204
+    // In order to not hit the rate limits and respect their rate limiting rule.
+    // We are going to delay every request by one second.
+
+    while (m_timer.elapsed(true) < kDelayDuration) {
```

Comment:
This is a hoot loop that burns CPU. Even worse than sleeping.

Reply:
Ah, yes you are right. Thanks for the feedback. 

Now I tried to solve this issue by extract the time passed until we reach to the next request from 1 second.